Compilation error when build with mingw on Windows

Part of compile log (for more info see https://github.com/msys2/MINGW-packages/pull/9612):
```sh
  In file included from src/printutils.h:9,
                   from src/qtgettext.h:16,
                   from src/ProgressWidget.h:3,
                   from src/ProgressWidget.cc:1:
  D:/a/_temp/msys/msys64/mingw64/include/boost/system/detail/snprintf.hpp: In function 'void boost::system::detail::snprintf(char*, std::size_t, const char*, ...)':
  D:/a/_temp/msys/msys64/mingw64/include/boost/system/detail/snprintf.hpp:57:10: error: 'libintl_vsnprintf' is not a member of 'std'; did you mean 'libintl_vsnprintf'?
     57 |     std::vsnprintf( buffer, len, format, args );
        |          ^~~~~~~~~
  D:/a/_temp/msys/msys64/mingw64/include/libintl.h:379:12: note: 'libintl_vsnprintf' declared here
    379 | extern int vsnprintf (char *, size_t, const char *, va_list);
        |            ^~~~~~~~~
```

--- a/src/printutils.h
+++ b/src/printutils.h
@@ -6,8 +6,17 @@
 #include <boost/format.hpp>
 #include <boost/algorithm/string.hpp>
 #include <utility>
+
 #include <libintl.h>
+// Undefine some defines from libintl.h to presolve
+// some collisions in boost headers later
+#if defined snprintf
 #undef snprintf
+#endif
+#if defined vsnprintf
+#undef vsnprintf
+#endif
+
 #include <locale.h>
 #include "AST.h"
 #include <set>

